llvm/lib/CodeGen/SelectionDAG/SDNodeDbgValue.h

```diff
@@ -30,10 +30,13 @@ class raw_ostream;
 class SDDbgOperand {
 public:
   enum Kind {
-    SDNODE = 0,  ///< Value is the result of an expression.
-    CONST = 1,   ///< Value is a constant.
-    FRAMEIX = 2, ///< Value is contents of a stack location.
-    VREG = 3     ///< Value is a virtual register.
+    SDNODE = 0,        ///< Value is the result of an expression.
+    CONST = 1,         ///< Value is a constant.
+    FRAMEIX = 2,       ///< Value is contents of a stack location.
+    VREG = 3,          ///< Value is a virtual register.
+    ENTRYVALUEREG = 4, ///< Value is a physical register that should be
```

Comment:
Do we really need a new Kind or would it be sufficient to append the EntryValue to the `DIExpression` when we create the `SDDbgNode` and look for VREG+EntryValue in the code?

Reply:
I tried this and it worked. So I am getting rid of this code. Great idea!